File: WorkerNode/src/main/java/com/yahoo/sdvornik/sorter/InsertionSort.java
package com.yahoo.sdvornik.sorter;

/**
 * Implementation of InsertionSort algorithm.
 */
public final class InsertionSort extends Sorter {

    /**
     * Ctor.
     * @param longArr
     */
    public InsertionSort(long[] longArr) {
        super(longArr);
    }

    public long[] sort(){
        for(int j = 1; j< longArr.length; ++j){
            long key = longArr[j];
            int i = j - 1;
            while (i >= 0 && longArr[i] > key){
                longArr[i + 1] = longArr[i];
                --i;
            }
            longArr[i + 1] = key;
        }
        return longArr;
    }
}

File: worker/src/main/kotlin/com/yahoo/sdvornik/sorter/InsertionSort.kt
package com.yahoo.sdvornik.sorter

/**
 * @author Serg Dvornik <sdvornik@yahoo.com>
 */
class InsertionSort
/**
 * Ctor.
 * @param longArr
 */
(longArr: LongArray) : Sorter(longArr) {

    override fun sort(): LongArray {
        for (j in 1..longArr.size - 1) {
            val key = longArr[j]
            var i = j - 1
            while (i >= 0 && longArr[i] > key) {
                longArr[i + 1] = longArr[i]
                --i
            }
            longArr[i + 1] = key
        }
        return longArr
    }
}